test/test_build.jl

```diff
@@ -59,6 +59,9 @@ end
             pgfsave("$tmp.tex", a)
             @test is_tex_document("$tmp.tex")
             println(read("$tmp.tex", String))
+            # test with filename::String{SubString}
+            pgfsave(split("foo|$tmp-2.tex", '|')[2], a)
```

Comment:
I would recommend that you save an `svg` or a `pdf`, so that the a larger part of the chain of functions (`run_latex_once` etc) that were modified by this PR gets called.

Don't worry about CI overhead, it is rather minor.